### DISCUSSION TITLE
Question regarding #224

### DISCUSSION THREAD

**Discussion by FatimaArshad-DS** [ModECI/MDF#235] [Q&A]
In SimpleExample.ipynb linear regression is implemented. I am trying to extend it to incorporate logistic regression. Do I have to implement it myself or use built-in MDF functions? 

For former Ill dig deeper into how it is implemented in MDF since I'm getting an error :D 

This is my graph:
<img width="464" alt="Screenshot 2022-04-04 at 10 55 10 PM" src="https://user-images.githubusercontent.com/78070220/161603221-da887a7d-88ba-4573-a3db-857f127c184f.png">

And Im getting this error on it. 
Error! Could not evaluate expression [1 /(1 + exp((-1) * gain *( input_val + bias) + offset) )] with params [input_val=4.2, gain=1.2, bias=3, offset=3, logistic_regression=0], returned [1 /(1 + exp((-1) * gain *( input_val + bias) + offset) )] which is a <class 'str'>

**Comment by FatimaArshad-DS** [comment]
This is because of exp. There is no built-in function for exponential and so it is treated like a string

**Comment by pgleeson** [comment]
@FatimaArshad-DS Yes, the issue is that the `exp` in the expression for logistic_regression is not being parsed correctly. In future versions of MDF if should be, but we'll need to figure out whether this should be mapped to math.exp(), numpy.exp(), torch.exp()  etc. (dependent on whether the parameter arguments are scalar or array/tensor ).

For now, you could add a new intermediate parameter (exp_val) alog the lines of this:
https://github.com/ModECI/MDF/blob/eb318dc1f5aef7637da8e50b52c86afff1489789/examples/MDF/abcd.py#L75-L85
and then use that parameter in the final expression for logistic_regression.
